nixos/modules/services/web-apps/invidious.nix

```diff
@@ -171,6 +187,17 @@ in
       '';
     };
 
+    hmacKeyFile = lib.mkOption {
+      type = types.nullOr types.path;
+      default = null;
+      description = lib.mdDoc ''
+        A path to a file containing the `hmac_key`. If `null`, a key will be generated automatically on first
+        start.
+
+        It overrides any `hmac_key` specified in {option}`services.invidious.settings`.
```

Comment:
This seems a bit wrong, defaults should never override explicit user settings. It should be this order from high to low:
- `hmacKeyFile`
- `extraSettingsFile`
- `settings` (though we should maybe complain about it not being safe if users specify the `hmac_key` here)
- An autogenerated default

I'm not sure if that's currently the case.

Reply:
I've changed the module to use this order now, keeping the `database.passwordFile` in between `settings` and `extraSettingsFile`.